### ISSUE TITLE
PubSup child module deployment may lead to Error 400: regions not allowed by the project's organization policy

### ISSUE THREAD

**Issue by FLouisy** [BrunoReboul/terraform-google-realtime-asset-monitor#55]
When deploying a pub/sub topic (module.realtime-asset-monitor.module.slos.google_pubsub_topic.alerting), this error raised:

`Error: Error creating Topic: googleapi: Error 400: The requested message storage policy contains the following GCP regions that are not allowed by the project's organization policy: europe-central2, europe-north1, europe-west3. The organization policy may be updated by your project or organization administrator.`

These parameters are defined in the terraform variables, with their default value.

```
variable "pubsub_allowed_regions" {
  type = list(string)
  default = ["europe-central2", "europe-north1", "europe-west1", "europe-west3", "europe-west4"]
}
```

Link: https://github.com/BrunoReboul/terraform-google-realtime-asset-monitor/blob/main/variables.tf

It is possible to change these variables in tfvars file.

However during calls to child RAM modules, as here for "slos" which deploys the alerting topic, these variables are not propagated.

```
resource "google_pubsub_topic" "alert" {
  project = var.project_id
  name = var.alerting_topic_name
  message_storage_policy {
    allowed_persistence_regions = var.pubsub_allowed_regions
  }
}
```

Link: https://github.com/BrunoReboul/terraform-google-realtime-asset-monitor/blob/main/modules/slos/main.tf

Values ​​used are those defined by default in the module, instead of updated ones:
https://github.com/BrunoReboul/terraform-google-realtime-asset-monitor/blob/main/modules/slos/variables.tf

Terraform does not really have global vars management, if we do not pass them to the module call, it uses those of the child module.
The call is here:

```
module "slos" {
  # Create SLOs once the log based metrics have been created
  depends_on = [module.metrics]
  source="./modules/slos"
  project_id = var.project_id
  notification_channels = var. notification_channels
  ram_e2e_latency = var.ram_e2e_latency
}
```

Link: https://github.com/BrunoReboul/terraform-google-realtime-asset-monitor/blob/main/main.tf

We should add this var to call:

`pubsub_allowed_regions = var.pubsub_allowed_regions`

**Maybe we have this kind of behavior on other child modules variables.**

**Comment by BrunoReboul** [comment]
Good catch !! thanks :-)

- removed all default values for this setting in the child modules
- added the missing setting propagation as suggested

**Comment by BrunoReboul** [comment]
Released as [v0.5.3](https://github.com/BrunoReboul/terraform-google-realtime-asset-monitor/releases/tag/v0.5.3)
